app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala

```diff
@@ -42,8 +43,9 @@ class CustomsNotificationService @Inject()(logger: NotificationLogger,
   def handleNotification(xml: NodeSeq, callbackDetails: DeclarantCallbackData, metaData: RequestMetaData)(implicit hc: HeaderCarrier): Future[Boolean] = {
     gaConnector.send("notificationRequestReceived", s"[ConversationId=${metaData.conversationId}] A notification received for delivery")
 
-    val value = hc.headers.seq.map(a => Header(a._1, a._2))
-    val clientNotification = ClientNotification(ClientSubscriptionId(UUID.fromString(metaData.clientId)), Notification(ConversationId(metaData.conversationId), value, xml.toString, MimeTypes.XML), None)
+    val headers = metaData.mayBeBadgeId.fold(Seq.empty[Header])(id => Seq(Header(X_BADGE_ID_HEADER_NAME, id)))
+
+    val clientNotification = ClientNotification(ClientSubscriptionId(UUID.fromString(metaData.clientId)), Notification(ConversationId(metaData.conversationId), headers, xml.toString, MimeTypes.XML), None)
```

Comment:
It is better to push this code "ClientSubscriptionId(UUID.fromString(metaData.clientId)" inside metadata and later on we can store the domain objects in the metadata instead of string.

Reply:
I will address that in a separate PR